File: aula2/Personagem.py
# Projeto IF Quest
# Aula 2 - POO

# Classe com atributos e comportamentos de cada pesonagem
class Personagem:
    def __init__(self):
        self.nome = ""
        self.vida = 100
        self.forca = 10


    def receberDano(self, dano):
        self.vida -= dano
        print(f"{self.nome} sofreu {dano} de dano.")


    def estaVivo(self):
        return self.vida > 0


    def ficha(self):
        return f"Nome: {self.nome}\nVida: {self.vida}\nForça: {self.forca}"


    def atacar(self, alvo):
        print(f"{self.nome} ataca {alvo.nome} com {self.forca} de força.")
        alvo.receberDano(self.forca)



# Programa principal
if __name__ == "__main__":

    # Criando o herói
    heroi = Personagem()
    heroi.nome = "Scribble"
    #heroi.vida = 100 # já está definido com esse valor inicialmente
    #heroi.forca = 10 # já está definido com esse valor inicialmente

    # Criando o chefe
    chefe = Personagem()
    chefe.nome = "Dragão Azul de Olhos Vermelhos"
    chefe.vida = 200
    chefe.forca = 20

    # Ficha antes da batalha
    print("Ficha do Herói antes da Batalha:")
    print(heroi.ficha())
    print("------------------------------------\n")
    print("Ficha do Chefe antes da Batalha:")
    print(chefe.ficha())
    print("------------------------------------\n")

    print("---------- Início da Batalha ---------\n")


    # Simule uma batalha: cada personagem ataca o outro alternadamente até que um deles morra.
    while heroi.estaVivo() and chefe.estaVivo():
        
        # Ataque do herói
        heroi.atacar(chefe)

        if not chefe.estaVivo():
            print(chefe.nome + " está morto.")
            print(heroi.nome + " venceu a batalha!")
            break
        
        print()

        # Ataque do chefe
        chefe.atacar(heroi)

        if not heroi.estaVivo():
            print(heroi.nome + " está morto.")
            print(chefe.nome + " venceu a batalha!")
            break


    print()


    # Ficha depois da batalha
    print("Ficha do Herói depois da Batalha:")
    print(heroi.ficha())
    print("------------------------------------\n")
    print("Ficha do Chefe depois da Batalha:")
    print(chefe.ficha())
    print("------------------------------------\n")

    print("---------- Fim da Batalha ---------\n")

File: aula2/Personagem.java
// Projeto IF Quest
// Aula 2 - POO


// Classe com atributos e comportamentos de cada pesonagem
public class Personagem {
    String nome;
    int vida;
    int forca;


    void receberDano(int dano) {
        vida -= dano;
        System.out.println(nome + " sofreu " + dano + " de dano.");
    }

    boolean estaVivo() {
        return vida > 0;
    }

    String ficha() {
        return "Nome: " + nome + "\nVida: " + vida + "\nForça: " + forca;
    }

    void atacar(Personagem alvo) {
        System.out.println(nome + " ataca " + alvo.nome + " com " + forca + " de força.");
        alvo.receberDano(forca);   
    }
}

// Classe principal
class Main {
    public static void main(String[] args) {
        
        // Criando o herói
        Personagem heroi = new Personagem();
        heroi.nome = "Scribble";
        heroi.vida = 100;
        heroi.forca = 10;

        // Criando o chefe
        Personagem chefe = new Personagem();
        chefe.nome = "Dragão Azul de Olhos Vermelhos";
        chefe.vida = 200;
        chefe.forca = 20;

        // Ficha antes da batalha
        System.out.println("Ficha do Herói antes da Batalha:");
        System.out.println(heroi.ficha());
        System.out.println("------------------------------------\n");
        System.out.println("Ficha do Chefe antes da Batalha:");
        System.out.println(chefe.ficha());
        System.out.println("------------------------------------\n");

        System.out.println("---------- Início da Batalha ---------\n");


        // Simule uma batalha: cada personagem ataca o outro alternadamente até que um deles morra.
        while (heroi.estaVivo() && chefe.estaVivo()) {
            
            // Ataque do herói
            heroi.atacar(chefe);

            if (!chefe.estaVivo()) {
                System.out.println(chefe.nome + " está morto.");
                System.out.println(heroi.nome + " venceu a batalha!");
                break;
            }

            System.out.println();


            // Ataque do chefe
            chefe.atacar(heroi);

            if (!heroi.estaVivo()) {
                System.out.println(heroi.nome + " está morto.");
                System.out.println(chefe.nome + " venceu a batalha!");
                break;
            }
        }

        System.out.println();

        // Ficha depois da batalha
        System.out.println("Ficha do Herói depois da Batalha:");
        System.out.println(heroi.ficha());
        System.out.println("------------------------------------\n");
        System.out.println("Ficha do Chefe depois da Batalha:");
        System.out.println(chefe.ficha());
        System.out.println("------------------------------------\n");

        System.out.println("---------- Fim da Batalha ---------\n");
    
    }
}